rust_src/src/buffers.rs

```diff
@@ -309,6 +311,24 @@ impl LispBufferRef {
         self.zv = charpos;
         self.zv_byte = byte;
     }
+
+    #[inline]
+    pub fn set_syntax_table(&mut self, table: LispCharTableRef) {
+        self.syntax_table = LispObject::from(table).to_raw();
+    }
+
+    /// Set whether per-buffer variable with index IDX has a buffer-local
+    /// value in buffer.  VAL zero means it does't.
+    // Similar to SET_PER_BUFFER_VALUE_P macro in C
+    #[inline]
+    pub fn set_per_buffer_value_p(&mut self, idx: usize, val: c_uchar) {
+        unsafe {
+            if idx >= last_per_buffer_idx {
+                emacs_abort()
```

Comment:
I think we use `panic!()`

Reply:
Replaced. Sorry. I forgot about this from the Porting Tips

Reply:
The code is currently inconsistent. We should decide on which we prefer and do a quick cleanup.